### PR TITLE
IVR: unify the canvas position label

### PR DESCRIPTION
**Pull request by davidpmccormick**
- For #12986

## What does this change?

The "3/48" canvas position indicator was built four different ways. `ViewerTopBar` and `ViewerBottomBar` both render it, `PaginatedItemViewer` passes it to `IIIFItem` as a fallback caption, and `VirtualizedImageViewer` did the same using a 0-based `react-window` row index. Three of the four recalculated the total from `transformedManifest.canvases.length` even though `totalCanvases` is already on `ItemViewerContext`.

A new `CanvasPositionIndicator` owns both where the two values come from and the format they're rendered in. It reads `query.canvas` and `totalCanvases` off context itself rather than taking them as props, so callers can't substitute a different source. The format lives in one private hook, returned pre-split at the point the top bar needs an element boundary for its `active-index` test id — so a format change can't update the markup and quietly leave the string version behind.

`VirtualizedImageViewer`'s `titleOverride` is **deleted rather than corrected**. Tests-first turned up that the prop was unreachable there: `IIIFItem` only reads it as the audio player's fallback title, and that viewer never renders audio. See the inline comments for the full derivation and for what the characterisation test actually rendered before the deletion.

Two dead fallbacks in `ViewerTopBar` went with it (`query.canvas || 0` and `canvases?.length || ''`) — details inline.

`IIIFSearchWithin.tsx:128` renders the same two values as `Found on image 3 / 48 (page 3)`. Left alone deliberately: different wording and spacing, and unifying it would change visible text in search results, so it wants its own review and probably a design opinion.

## How to test

No user-visible change is expected, so this is mostly a confirm-nothing-moved exercise.

With the IVR toggle on, compare against prod:
- A multi-canvas image work, e.g. [b/a2239muq](https://www-dev.wellcomecollection.org/works/a2239muq/items?canvas=15) — the top bar should read `15/80 page 15` exactly as prod does, and paging should keep it in step.
- A non-image work (video/audio/born-digital) — the bottom bar's `Previous 2/3 Next` counter should be unchanged.
- An audio work whose canvas has no label of its own — the audio player's caption should show the 1-based position (`2/2`, not `0/2`).

Without the toggle, nothing should have changed from prod.

## Have we considered potential risks?

Low risk — this moves where an existing string is built, and the values it's built from are the same ones the bottom bar already used.

Two things worth a reviewer's eye:

The deletion rests on an invariant that spans four files rather than anything local, so `IIIFViewer.test.tsx` pins it by driving a real audio manifest through the real `IIIFViewer`. If you disagree that deleting beats fixing, that's the decision to push back on.

The bottom bar's counter gains a nested span, which is the one DOM change. It breaks up the text for testing-library's `getByText`, so one assertion moved to `toHaveTextContent`. Playwright is unaffected — its text engine matches the smallest element containing the text, so the outer span still resolves uniquely for `5/27`, and `active-index` still holds only the bare number.

Merge-tested against the two sibling PRs on this issue, #13409 and #13412: both merge clean pairwise and combined, with the full `content/webapp` suite passing on the combined tree (83 suites, 641 tests).

### CODE REVIEW

**Review by davidpmccormick — COMMENTED** [review]
_Written by Claude_

Notes I (Claude) trimmed out of the code on a "this is reasoning about the change, not documentation of it" pass. Recording it here so the derivations are on the record without the files carrying them permanently.

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { FunctionComponent } from 'react';
+
+import { useItemViewerContext } from '@weco/content/contexts/ItemViewerContext/refactored';
+
+// The current canvas's position within the manifest - "3/48" - is shown in the
+// top bar's page indicator, the bottom bar's canvas navigation, and as the
+// fallback caption for an audio item with no label of its own. This module owns
+// both where the two values come from and the format they're rendered in.
```
Comment (davidpmccormick) [inline content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx]: _Written by Claude_

Background on why this module exists at all.

Before this, each of the three places that showed the position built the string itself, reading the position and total from whichever source was nearest to hand — `ViewerTopBar`, `PaginatedItemViewer` and `VirtualizedImageViewer` all recalculated the total from `transformedManifest.canvases.length` even though `totalCanvases` was already on `ItemViewerContext`. Only `ViewerBottomBar` used the context value.

That's how the four drifted apart, and it's why this now owns both the values and the format rather than just offering a helper the call sites may or may not reach for.

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+   * carry a given id - the top bar and bottom bar both render one, and a
+   * shared id would match twice.
+   */
+  positionTestId?: string;
+};
+
+/**
+ * The current canvas's position, for display. Reads the values off
+ * ItemViewerContext rather than taking them as props, so callers can't
+ * substitute a different source.
```
Comment (davidpmccormick) [inline content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx:38]: _Written by Claude_

The source that actually got substituted in practice was `VirtualizedImageViewer` passing its 0-based `react-window` row index, plus three call sites recalculating the total off the manifest.

Reading straight off context is what makes that class of drift impossible rather than merely discouraged — which is the reason for the slightly unusual choice of a component that takes no data props.

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/VirtualizedImageViewer.tsx [hunk 3]
```diff
@@ -74,11 +74,14 @@ const ItemRenderer = memo(({ style, index, data }: ItemRendererProps) => {
                   data-testid="image-item"
                   $isFirstItemRestricted={!!isFirstItemRestricted}
                 >
+                  {/*
+                    No titleOverride: IIIFItem only reads it for audio, and
+                    this viewer never renders audio - see IIIFViewer.test.tsx.
+                  */}
```
Comment (davidpmccormick) [inline content/webapp/views/pages/works/work/IIIFViewer/refactored/VirtualizedImageViewer.tsx]: _Written by Claude_

Full derivation, since the comment only asserts it:

1. `refactored/IIIFItem/index.tsx:439` — `titleOverride` reaches exactly one consumer, `AudioPlayer`'s `title` prop. Worth noting that's a visible `<figcaption>`, not a title attribute, so the stakes were a bit higher than the issue assumed.
2. `MainViewer.tsx:23` — only picks `VirtualizedImageViewer` when `hasOnlyRenderableImages` is true.
3. `IIIFViewer.tsx:311` — `hasOnlyRenderableImages = !hasNonImagesOrOriginals(canvases)`.
4. `utils/iiif/v3/index.ts:796` — `hasNonImagesOrOriginals` returns true for any canvas with a non-`Image` painting/rendering/supplementing body, or any `original`. `Sound`, `Audio` and `Choice` are all non-`Image`.

Chain them: an audio canvas always routes to `PaginatedItemViewer`, so this viewer only ever renders `Image` items — and the image branch ignores the prop.

What the characterisation test found before the deletion, since the issue asked to confirm the off-by-one before touching anything: the value here was `` `${index}/${canvases.length}` `` with `index` 0-based, so rendering this viewer directly with a label-less audio canvas at `canvas=2` produced a caption reading `0/2`. The off-by-one was real, just unreachable. That's what tipped it from "fix" to "delete" — there was no correct value to give a prop nothing reads.

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/IIIFViewer.test.tsx [hunk 4]
```diff
@@ -161,4 +161,34 @@ describe('IIIFViewer', () => {
       expect(fetchMock).not.toHaveBeenCalled();
     });
   });
+
+  // VirtualizedImageViewer relies on this: it passes no titleOverride because
+  // IIIFItem only reads it for audio. MainViewer.test.tsx covers the other
+  // direction, given the flag.
```
Comment (davidpmccormick) [inline content/webapp/views/pages/works/work/IIIFViewer/refactored/IIIFViewer.test.tsx:247]: _Written by Claude_

Why this test is here rather than trusting the reasoning.

The invariant it pins spans four files (see the `VirtualizedImageViewer` comment). None of them would obviously break on its own — someone could edit `hasNonImagesOrOriginals`, or the `hasOnlyRenderableImages` derivation, or `MainViewer`'s branch, each edit looking entirely reasonable locally, and the only symptom would be a missing audio caption in a viewer nobody was looking at.

So it drives a real audio manifest through the real `IIIFViewer` rather than mocking the flag, and fails if any link in the chain changes. `MainViewer.test.tsx` already covers the reverse direction given the flag, so I didn't duplicate the image-only converse here.

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/PaginatedItemViewer.test.tsx [hunk 5]
```diff
@@ -149,6 +149,30 @@ describe('PaginatedItemViewer', () => {
     ).toBeInTheDocument();
   });
 
+  // titleOverride is IIIFItem's fallback for an item whose canvas has no label
+  // of its own - getFileLabel treats "-" as no label - and only the audio
+  // player reads it.
```
Comment (davidpmccormick) [inline content/webapp/views/pages/works/work/IIIFViewer/refactored/PaginatedItemViewer.test.tsx:154]: _Written by Claude_

After the `VirtualizedImageViewer` deletion this is the only remaining `titleOverride` call site, which is why it's worth pinning here rather than leaving to the component tests. I left that fact out of the comment itself because it goes stale the moment anyone adds another one.

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/ViewerTopBar.tsx [hunk 6]
```diff
@@ -375,8 +375,7 @@ const ViewerTopBar: FunctionComponent<ViewerTopBarProps> = ({
         <MiddleZone className="viewer-desktop">
           {shouldShowPageIndicator && (
             <>
-              <span data-testid="active-index">{`${query.canvas || 0}`}</span>
-              {`/${canvases?.length || ''}`}{' '}
+              <CanvasPositionIndicator positionTestId="active-index" />{' '}
```
Comment (davidpmccormick) [inline content/webapp/views/pages/works/work/IIIFViewer/refactored/ViewerTopBar.tsx:311]: _Written by Claude_

The two fallbacks this replaces — `query.canvas || 0` and `canvases?.length || ''` — were both unreachable, so dropping them is behaviour-preserving rather than a tightening:

- The total can't be empty here. `shouldShowPageIndicator` requires `hasMultipleCanvases`, which is `totalCanvases > 1`, so it's always at least 2 by the time this renders.
- The position can't be falsy. `maybeNumberCodec` maps unparseable *and* zero values to `undefined` (`parseInt(p, 10) || undefined`), and `IIIFViewer.tsx:264` defaults `canvas = 1`. `NaN` isn't reachable either.

The existing `canvas: 0` and `canvas: 9999` edge-case tests in `ViewerTopBar.test.tsx` still pass unchanged — `{0}` renders as `"0"` just as `` `${0 || 0}` `` did.

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.test.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+} from './CanvasPositionIndicator';
+
+const threeCanvases = createMockManifest({
+  canvases: [createMockCanvas(), createMockCanvas(), createMockCanvas()],
+});
+
+const render = (
+  ui: React.ReactElement,
+  contextProps: Partial<ItemViewerContextProps> = {}
+) =>
```
[the remark this answers is not in the document]
Comment (davidpmccormick) [reply content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.test.tsx:23]: Autofix broke the build so I went with the other suggestion (destructuring `ReactElement`)

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/VirtualizedImageViewer.tsx (hunk 3 above)
Comment (davidpmccormick) [reply content/webapp/views/pages/works/work/IIIFViewer/refactored/VirtualizedImageViewer.tsx]: I'm not convinced even the much shorter comment that Claude has left here is worth keeping in the codebase

**Review by rcantin-w — COMMENTED** [review]
Unrelated to this PR but I'm starting to wonder if we want a sub-folder for all these smaller components 😂 (I've created more too)

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.test.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  });
+
+  it('shows an out-of-range canvas number as-is rather than clamping it', () => {
+    const { container } = render(<CanvasPositionIndicator />, {
+      query: createMockQuery({ canvas: 9999 }),
+    });
+
+    expect(container).toHaveTextContent('9999/3');
+  });
+});
```
Comment (rcantin-w) [inline content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.test.tsx:63]: I guess we want that? 😄

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx (hunk 1 above)
Comment (rcantin-w) [inline content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx]: I'm not sure the comment is useful to have?

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  const { query, totalCanvases } = useItemViewerContext();
+
+  return { position: `${query.canvas}`, suffix: `/${totalCanvases}` };
+}
+
+/**
+ * The current canvas's position as a plain string, e.g. "3/48", for the places
+ * that need a value rather than markup - the audio player takes its fallback
+ * title as a string prop.
+ */
```
Comment (rcantin-w) [inline content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx]: Again here and along this whole file, I'm not sure we need to point out specific ways it's used, it makes me think of this rule I added to my instructions:

> For hooks/plain functions (not components), prefer JSDoc comments stating only what something is for, not the backstory of how it got that way

What do you think? Might help with amount of text.

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/ViewerBottomBar.test.tsx [hunk 10]
```diff
@@ -133,7 +133,9 @@ describe('ViewerBottomBar navigation (non-image works)', () => {
       },
     });
 
-    expect(screen.getByText('2/3')).toBeInTheDocument();
+    // CanvasPositionIndicator splits the position into its own span so the top
+    // bar can put a test id on it, which breaks up the text for getByText.
```
Comment (rcantin-w) [inline content/webapp/views/pages/works/work/IIIFViewer/refactored/ViewerBottomBar.test.tsx]: can be removed?

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/VirtualizedImageViewer.tsx (hunk 3 above)
Comment (rcantin-w) [reply content/webapp/views/pages/works/work/IIIFViewer/refactored/VirtualizedImageViewer.tsx]: I'm wondering if we shouldn't have Audio get his own titleOverride instead of prop drilling it, but idk how doable that would be?

**Review by rcantin-w — APPROVED** [review]
Looks good! Most of my comments are about the about of text added for comments I'm not _sure_ we need. Then maybe an idea for `titleOverride` I'm not even sure is doable 😄 

But yay 🧹

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx (hunk 1 above)
Comment (davidpmccormick) [reply content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx]: agree – some comments seemingly escaped my Big Cull

File: content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx (hunk 9 above)
Comment (davidpmccormick) [reply content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx]: I felt this way completely until the roundtable, then I wondered whether these are potentially more useful for agents than for us? But I'm happy to get rid

Comment (rcantin-w) [reply content/webapp/views/pages/works/work/IIIFViewer/refactored/CanvasPositionIndicator.tsx]: I don't think it's always a no, I think it can be useful to have history when it's a bit weird/specific looking, but this feels more like "where it's used" thing? which is likely to change?
